File: src/main/java/net/thenamedev/legendarena/utils/Cooldown.java
package net.thenamedev.legendarena.utils;

import org.bukkit.*;
import org.jetbrains.annotations.*;

/**
 * Code stolen from: http://hastebin.com/noqemohuda.java
 */
public class Cooldown {

    private int seconds;
    private int total;
    private long start;

    public Cooldown(int seconds) {
        this.seconds = seconds;
        this.total = seconds;

        this.start = System.currentTimeMillis();
    }

    public boolean done() {
        return ((System.currentTimeMillis() - start) > seconds * 1000);
    }

    private String remainingSeconds() {
        long number = System.currentTimeMillis() - start;

        return String.format("%.1f", seconds - (number / 1000.0));
    }

    @NotNull
    public String getTimeRemaining() {
        return PluginUtils.msgNormal + "Cooldown: " + ChatColor.YELLOW + remainingSeconds() + ChatColor.GRAY + "/" + ChatColor.YELLOW + total + "s";
    }

}

File: src/main/java/legendapi/utils/Cooldown.kt
package legendapi.utils

import org.bukkit.ChatColor

public class Cooldown {

    private val seconds: Double
    private val start: Long

    public constructor(seconds: Double) {
        this.seconds = seconds
        this.start = System.currentTimeMillis()
    }

    public fun done(): Boolean {
        if(java.lang.String.format("%.1f", seconds - ((System.currentTimeMillis() - start) / 1000.0)) == "0.0") return true //fucking idiotic cooldown system bugs
        return ((System.currentTimeMillis() - start) > seconds * 1000)
    }

    public fun remainingSeconds(): String {
        return java.lang.String.format("%.1f", seconds - ((System.currentTimeMillis() - start) / 1000.0))
    }

    public fun getTimeRemaining(): String {
        return ChatUtils.getCustomMsg("Cooldown") + remainingSeconds() + ChatColor.GRAY + "/" + ChatColor.BLUE + seconds + "s"
    }

}